src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -735,40 +735,42 @@ internal static void UpdateLayerBorder(this CoreAnimation.CALayer layer, IButton
 			return null;
 		}
 
-		internal static UIView? FindNextView(this UIView view, UIView superView, Func<UIView, bool> isValidType)
+		internal static UIView? FindNextView(this UIView view, UIView containerView, Func<UIView, bool> isValidType)
 		{
-			var passedOriginal = false;
+			UIView? nextView = null;
 
-			var nextView = superView.FindNextView(view, ref passedOriginal, isValidType);
+			while (view != containerView && view is not null)
+			{
+				var siblings = view.Superview.Subviews;
+				nextView = view.FindNextView(siblings.IndexOf(view) + 1, isValidType);
+
+				if (nextView is not null)
+					break;
+
+				view = view.Superview;
+			}
 
 			// if we did not find the next view, try to find the first one
-			nextView ??= superView.FindNextView(null, ref passedOriginal, isValidType);
+			nextView ??= containerView.Subviews[0].FindNextView(0, isValidType);
 
 			return nextView;
 		}
 
-		static UIView? FindNextView(this UIView view, UIView? origView, ref bool passedOriginal, Func<UIView, bool> isValidType)
+		static UIView? FindNextView(this UIView view, int index, Func<UIView, bool> isValidType)
 		{
-			foreach (var child in view.Subviews)
+			// search through the view's siblings and traverse down their branches
+			var siblings = view.Superview.Subviews;
+			for (int i = index; i < siblings.Length; i++)
 			{
-				if (isValidType(child))
+				if (siblings[i].Subviews.Length > 0)
```

Comment:
This indexes into `siblings[i]` a couple times. Could you create `var sibling = siblings[i];` at the top of the loop and use `sibling` instead? This way it will skip the array bounds checks, etc. that happen when you index into an array.

Reply:
Ah good call yeah I can add that in!

Reply:
Updated accordingly @jonathanpeppers